sound/soc/sof/pcm.c

```diff
@@ -522,6 +529,11 @@ static int sof_pcm_close(struct snd_pcm_substream *substream)
 		 */
 	}
 
+	/* update active pcm counts */
+	sdev->pcm_active--;
+	if (spcm->stream[substream->stream].avail_at_lp)
+		sdev->lp_pcm_active--;
```

Comment:
and here? locking seems required for refcounting?

Reply:
Same as at .open(), we don't need mutex or locking.